src/vs/workbench/contrib/files/browser/views/explorerViewer.ts

```diff
@@ -275,6 +278,60 @@ export class FilesRenderer implements ICompressibleTreeRenderer<ExplorerItem, Fu
 	private _onDidChangeActiveDescendant = new EventMultiplexer<void>();
 	readonly onDidChangeActiveDescendant = this._onDidChangeActiveDescendant.event;
 
+	private readonly hoverDelegate = new class implements IHoverDelegate {
+
+		readonly placement = 'element';
+
+		get delay() {
+			return this.configurationService.getValue<number>('workbench.hover.delay');
+		}
+
+		constructor(
+			private readonly configurationService: IConfigurationService,
+			private readonly hoverService: IHoverService
+		) { }
+
+		showHover(options: IHoverDelegateOptions, focus?: boolean): IHoverWidget | undefined {
+			let element: HTMLElement;
+			if (options.target instanceof HTMLElement) {
+				element = options.target;
+			} else {
+				element = options.target.targetElements[0];
+			}
+
+			const child = element.children[0] as Element | undefined;
+			const childOfChild = child?.children[0] as HTMLElement | undefined;
+			let overflowed = false;
+			if (childOfChild && child) {
+				const width = child.clientWidth;
+				const childWidth = childOfChild.offsetWidth;
+				// Check if element is overflowing its parent container
+				overflowed = width <= childWidth;
+			}
+
+			const hasDecoration = element.classList.toString().includes('monaco-decoration-iconBadge');
+			// If it's overflowing or has a decoration show the tooltip
+			overflowed = overflowed || hasDecoration;
+			// TODO @lramos15 find a better way to do this, grabs the indent element to position hover above the label
+			const indentGuideElement = element.parentElement?.parentElement?.children[0] as HTMLElement | undefined;
+
+			if (!indentGuideElement) {
+				return;
+			}
+
+			return overflowed ? this.hoverService.showHover({
+				...options,
+				target: indentGuideElement,
+				compact: true,
+				additionalClasses: ['explorer-item-hover'],
+				skipFadeInAnimation: true,
+				showPointer: false,
+				forwardClickEvent: true,
+				hoverPosition: HoverPosition.RIGHT,
+			}, focus) : undefined;
```

Comment:
The pointer messes up here on hover, despite being passed through. Actually @TylerLeonhardt just made a change to allow passing in `IHoverOptions.container` which might help with this (update from main).

Reply:
Not sure what you mean by the pointer messes up here?

Reply:
`cursor: pointer` becomes `cursor: default` when the mouse is moved after the hover appears

Reply:
Fixed via CSS